Assets/Scripts/SS3D/Systems/Tile/TileMapCreator/AssetGridManager.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using UnityEngine;
+using UnityEngine.UI;
+
+/// <summary>
+/// Component, that changes GridLayoutGroup contstraintCount to fit it's width
+/// </summary>
+[RequireComponent(typeof(GridLayoutGroup))]
+public class AssetGridManager : MonoBehaviour
```

Comment:
put it into an appropriate namespace and document it a bit more, maybe with an example. You can describe how it act on the construction menu.